flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Table.java

```diff
@@ -89,6 +89,11 @@
 	 */
 	TableOperation getTableOperation();
 
+	/**
+	 * Returns the {@link TableEnvironment} of this table.
+	 */
+	TableEnvironment getTableEnvironment();
```

Comment:
Let's not add this method. My opinion is we should decouple `Table` from `TableEnvironment`. The only obstacle so far is only the `writeToSink` method.

Reply:
The conversions in `TableConversions.scala` need TableEnvironment.  I will put `TableConversions` in planner module for now and remove getTableEnvironment() in Table.